Assets/MixedRealityToolkit-Examples/Demos/Boundary/Scripts/BoundaryVisualizer.cs

```diff
@@ -0,0 +1,124 @@
+using UnityEngine.Experimental.XR;
+
+namespace Microsoft.MixedReality.Toolkit.Examples.Demos
+{
+    /// <summary>
+    /// Demo class to show different ways of using the boundary API.
+    /// </summary>
+    public class BoundaryVisualizer : MonoBehaviour
+    {
+        [SerializeField]
+        [Tooltip("Material used to draw the boundary geometry.")]
+        private Material boundsMaterial = null;
+
+        [SerializeField]
+        [Tooltip("Material used to draw the area outside of the boundary geometry.")]
+        private Material outOfBoundsMaterial = null;
+
+        [SerializeField]
+        [Tooltip("Material used to draw the area inside of the inscribed rectangle.")]
+        private Material inscribedRectangleMaterial = null;
+
+        /// <summary>
+        /// Boundary system implementation.
+        /// </summary
+        private IMixedRealityBoundarySystem boundaryManager = null;
+        private IMixedRealityBoundarySystem BoundaryManager => boundaryManager ?? (boundaryManager = MixedRealityManager.Instance.GetManager<IMixedRealityBoundarySystem>());
+
+        private void Start()
+        {
+            AddQuad();
+            AddIndicators();
+        }
+
+        /// <summary>
+        /// Displays the boundary as a quad primitive.
+        /// </summary>
+        private void AddQuad()
+        {
+            // Get the rectangular bounds.
+            Vector2 center = EdgeUtilities.InvalidPoint;
+            float angle = 0f;
+            float width = 0f;
+            float height = 0f;
+            if (!(bool)(BoundaryManager?.TryGetRectangularBoundsParams(out center, out angle, out width, out height)))
+            {
+                // No rectangular bounds.
+                return;
+            }
+            
+            // Render the rectangular bounds.
+            if (EdgeUtilities.IsValidPoint(center))
+            {
+                GameObject quad = GameObject.CreatePrimitive(PrimitiveType.Quad);
+                quad.transform.SetParent(transform);
+                quad.transform.Translate(new Vector3(center.x, 0.005f, center.y)); // Add fudge factor to avoid z-fighting
+                quad.transform.Rotate(new Vector3(90, -angle, 0));
+                quad.transform.localScale = new Vector3(width, height, 1.0f);
+                quad.GetComponent<Renderer>().sharedMaterial = inscribedRectangleMaterial;
+            }
+        }
+
+        /// <summary>
+        /// Displays the boundary as an array of spheres where spheres in the
+        /// bounds are a different color.
+        /// </summary>
+        private void AddIndicators()
+        {
+            // Get the rectangular bounds.
+            Vector2 centerRect = EdgeUtilities.InvalidPoint;
+            float angleRect = 0f;
+            float widthRect = 0f;
+            float heightRect = 0f;
+            bool haveBoundary = (bool)(BoundaryManager?.TryGetRectangularBoundsParams(out centerRect, out angleRect, out widthRect, out heightRect));
```

Comment:
Let's bail and throw an error if the manager is null instead.

Reply:
i wanted to show something when there was no boundary (or system) but at this point, if it will get the change approved.... fine

Reply:
Ah I see, Just curious. That makes sense then.